framework/math/unknown_manager/unknown_manager.h

```diff
@@ -187,7 +187,7 @@ class UnknownManager
   unsigned int AddUnknown(UnknownType unk_type, unsigned int dimension = 0);
 
   /// Maps the unknown's component within the storage of a node.
-  unsigned int MapUnknown(unsigned int unknown_id, unsigned int component = 0) const;
+  unsigned int MapUnknown(int unknown_id, unsigned int component = 0) const;
```

Comment:
just curious, why the change from `unsigned `int to `int ` for unknown_id? Do we allow those ids to be negative now?

Reply:
Because in almost every routine where `unknown_id` was used there was a check for a negative value